File: Orion/ClientApp/src/app/shared/responses/product/product-details.response.ts
import {PagedResponse} from "../../infrastructure/queries/pagination/paged-response.interface";
import {ProductDetails} from "../../models/product/product-details";
import {ProductPriceList} from "../../models/product/product-price-list";

export interface ProductDetailsResponse {
  productDetails: ProductDetails;
  productPrices: PagedResponse<ProductPriceList[]>;
}

File: Orion/Controllers/Responses/Product/ProductDetailsResponse.cs
using Orion.Application.Dtos.Product;
using Orion.Infrastructure.Queries.Pagination;
using Orion.Infrastructure.Responses;

namespace Orion.Controllers.Responses.Product
{
    public class ProductDetailsResponse
    {
        public ProductDetailsDto ProductDetails { get; set; }
        public PagedResponseEnvelope<ProductPriceListDto> ProductPrices { get; set; }

    }
}
